spec/helpers/nodeDefineJasmineUnderTest.js

```diff
@@ -16,7 +16,9 @@
       return path.join(__dirname, '../../', 'src/', file);
     });
 
-    const files = src_files.flatMap(g => glob.sync(g));
+    const files = src_files.flatMap(g =>
+      glob.sync(g, { windowsPathsNoEscape: true })
```

Comment:
Can you tell me more about what problem this solves? The tests work on my Windows machine without this change. And none of the globs contain backslashes. In any case, `windowsPathsNoEscape` has its share of rough edges and I'd rather find another way to solve the problem.

Reply:
`path.join()` returns paths using backslashes, so the result of:
```ts
  const src_files = ['core/**/*.js', 'version.js'].map(function(file) {
    return path.join(__dirname, '../../', 'src/', file);
  });
```
is 
```ts
  'C:\dev\jasmine\src\core\**\*.js',
  'C:\dev\jasmine\src\version.js'
```
`glob` [doesn't treat backslashes as separators by default](https://www.npmjs.com/package/glob#windows), so the result of globbing those paths are `[]` and `[ 'src\\version.js' ]`, respectively. The error reported is this:
```
C:\dev\jasmine> npm test

> jasmine-core@5.2.0 test
> grunt --stack execSpecsInNode

Running "execSpecsInNode" task
Error: Cannot find module 'src\version.js'
Require stack:
- C:\dev\jasmine\spec\helpers\nodeDefineJasmineUnderTest.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at C:\dev\jasmine\spec\helpers\nodeDefineJasmineUnderTest.js:24:7
    at Array.forEach (<anonymous>)
    at getSourceFiles (C:\dev\jasmine\spec\helpers\nodeDefineJasmineUnderTest.js:23:11)
    at C:\dev\jasmine\spec\helpers\nodeDefineJasmineUnderTest.js:28:3
    at Object.<anonymous> (C:\dev\jasmine\spec\helpers\nodeDefineJasmineUnderTest.js:32:3)
    at Module._compile (node:internal/modules/cjs/loader:1358:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'C:\\dev\\jasmine\\spec\\helpers\\nodeDefineJasmineUnderTest.js' ]
}
Warning: Task "execSpecsInNode" failed. Use --force to continue.
Error: Task "execSpecsInNode" failed.
    at Task.<anonymous> (C:\dev\jasmine\node_modules\grunt\lib\util\task.js:198:15)
    at Timeout._onTimeout (C:\dev\jasmine\node_modules\grunt\lib\util\task.js:234:33)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

Aborted due to warnings.
```

Reply:
Neat. The existing code works when the repo is checked out under c:\Users but not when it's checked out under c:\dev. I would not have expected that to make a difference. 

[
I've pushed a commit to main that fixes this without using windowsPathNoEscape](https://github.com/jasmine/jasmine/commit/3c4b73f136d0b7b10f49c2dfbe38b8e8e119260a).